client/python/openlineage/client/client.py

```diff
@@ -127,8 +105,150 @@ def filter_event(
                 return None
         return event
 
+    def emit(self, event: Event) -> None:
+        from typing import get_args
+
+        if type(event) not in get_args(Event):
+            msg = "`emit` only accepts RunEvent, DatasetEvent, JobEvent classes"
+            raise ValueError(msg)
+        if not self.transport:
+            log.error("Tried to emit OpenLineage event, but transport is not configured.")
+            return
```

Comment:
Probably it's better to raise exception in OpenLineageClient constructor if transport is not provided

Reply:
In OpenLineageClient constructor we are using `console` Transport when there is no transport provided and we emit a warning. What you propose would require changing that behaviour, am i correct? IMHO, we should not remove this `console` fallback but maybe there are some arguments for it?

Reply:
If no transport is explicitly set, leading to fallback to `ConsoleTransport`, is `not self.transport` case is even possible?

Reply:
Not unless explicitly remove or something breaks. I think it's more of a safety measure here.